src/client/utils/url.ts

```diff
@@ -102,7 +102,7 @@ export let getProxyUrl = function (url: string, opts?): string {
         return url;
 
     charset = charset || (parsedResourceType.isScript || parsedResourceType.isServiceWorker) &&
-        document[INTERNAL_PROPS.documentCharset];
+        self.document && document[INTERNAL_PROPS.documentCharset];
```

Comment:
This condition become large. Let's move it to the `getCharsetFromDocument (self, parsedResourceType)` function.